bbl_screen-patch/patches/printerui/qml/settings/HardwarePage.qml

```diff
@@ -400,44 +392,39 @@ Rectangle {
             Component {
                 id: choiceMenu
                 Choise {
-                    property var jsonKey: ""
-                    property var defaultSelection: "0"
-                    property var jsonval: DeviceManager.getSetting(jsonKey, defaultSelection)
+                    property var pressType: ""
+                    property var btn: ""
+                    property var defaultSelection
                     textFont: Fonts.body_24
                     listTextFont: Fonts.body_24
                     width: 115
-                    model: buttonActions.map(a => a.name)
-                    placeHolder: buttonActions[currentIndex].name
-                    currentIndex: buttonActions.findIndex(pair => pair.val == String(jsonval).charAt(0))
+                    model: X1Plus.GpioKeys.buttonActions.map(a => a.name)
+                    placeHolder:""
+                    currentIndex: defaultSelection
                     onCurrentTextChanged: { 
                         if(!down) return;
-                        dialog_jsonKey = jsonKey;
-                        var idx = buttonActions[currentIndex].val;
-                        if (idx == 6) {
-                            rect_isInputting_obj.setType = "macro";
-                            pathDialog("/mnt/sdcard/x1plus/macros/", "Enter macro filepath<br>(.gcode or .py)");
-                        } else if (idx == 1){
-                            rect_isInputting_obj.setType = "temp";
-                            pathDialog("nozzle 100", "ex: 'nozzle 100' or 'bed 45'<br>Replace value with temp (°C)");
-  
-                        } else {
-                            DeviceManager.putSetting(jsonKey, idx);
+                        let actionVal = X1Plus.GpioKeys.buttonActions.find(a => a.name === currentText).val;
+                        if (actionVal !== undefined) {
+                            console.log(`[x1p] {item.btn} {item.pressType} - {actionaVal}`);
+                            X1Plus.GpioKeys.updateButtonAction(item.btn, item.pressType, actionVal, {});
                         }
-                        X1PlusNative.system(`/etc/init.d/S82gpiokeys restart`);
                     }
+                    Binding on currentIndex{
+                        value: model.indexOf(X1Plus.GpioKeys.getActionText(btn, pressType))
+                    }                  
                 }
             }
-            
+          
             Loader {
                 Layout.fillWidth: true
                 sourceComponent: choiceMenu
-                onLoaded: { item.jsonKey = "cfw_power_short"; item.defaultSelection = "4" /* screen lock */; }
+                onLoaded: { item.btn = "cfw_power"; item.pressType = "shortPress";item.defaultSelection = X1Plus.GpioKeys.getDefault(item.btn,item.pressType); item.placeHolder =  X1Plus.GpioKeys.getActionText(item.btn, item.pressType)}
```

Comment:
hm I definitely would prefer that the defaultSelection and the placeHolder be bindings on the btn and pressType rather than repeating a bunch here